### PR TITLE
Workflow and build support for pip wheels

### PR DESCRIPTION
**Pull request by aaron-skydio**
See `documentation.rst` for usage description.  This primarily consists
of a workflow (manually triggered) to build wheels on GitHub Actions,
and supporting changes to our build to make this work, i.e. for the
right files to end up in the wheels in the right places (including
getting the README to look right on PyPI).

You can now:
- Build a wheel in a fresh environment, with basically whatever
  method you'd like (`python -m build`, `pip wheel`,
  `python setup.py bdist_wheel`).  Running this repeatedly to build
  multiple wheels also works, but may require cleaning the build
  directory depending on the method used (see the TODO on symenginepy
  in CMakeLists.txt)
- `pip install .` repeatedly to re-install symforce in the same
  environment
- `pip install -e .` repeatedly, manually cleaning the build
  (technically just the symenginepy build) between runs

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -86,6 +92,11 @@ set(SYMFORCE_EIGEN_TARGET "Eigen3::Eigen" CACHE STRING
 # Setup
 # ==============================================================================
 
+# If we're cross-compiling macOS arm64 wheels, set the architecture appropriately
+if("$ENV{CIBW_ARCHS_MACOS}" STREQUAL "arm64")
+  set(CMAKE_OSX_ARCHITECTURES arm64)
+endif()
```
Comment (hayk-skydio) [inline CMakeLists.txt:98]: at some point it might make sense to split out toolchain files

File: README.md [hunk 2]
```diff
@@ -2,7 +2,9 @@
     somewhere.  The closest thing is to add a #gh-light-mode-only to the target, which does this,
     but is kinda confusing -->
 ![SymForce](docs/static/images/symforce_banner.png#gh-light-mode-only)
+<!-- DARK_MODE_ONLY -->
 ![SymForce](docs/static/images/symforce_banner_dark.png#gh-dark-mode-only)
+<!-- /DARK_MODE_ONLY -->
```
Comment (hayk-skydio) [inline README.md:7]: Is this just a comment or does it do something?

File: setup.py [hunk 3]
```diff
@@ -6,19 +6,32 @@
...
 from setuptools import setup
 from setuptools.command.build_ext import build_ext
+from setuptools.command.egg_info import egg_info
 from setuptools.command.install import install
 
 SOURCE_DIR = Path(__file__).resolve().parent
 
 
+def symforce_version() -> str:
+    # We can't import the version here, so we have to do some text parsing
```
Comment (hayk-skydio) [inline setup.py:26]: turn this into docstring

File: setup.py [hunk 4]
```diff
@@ -88,12 +102,64 @@ def run(self) -> None:
...
         build_temp = Path(self.build_temp).resolve()
         dest_path = Path(self.get_ext_fullpath(ext.name)).resolve()  # type: ignore[attr-defined]
-        source_path = build_temp / "pybind" / self.get_ext_filename(ext.name)  # type: ignore[attr-defined]
         dest_directory = dest_path.parents[0]
         dest_directory.mkdir(parents=True, exist_ok=True)
-        self.copy_file(source_path, dest_path)
+        self.copy_file(extension_source_paths[ext.name], dest_path)  # type: ignore[attr-defined]
+
+
+class SymForceEggInfo(egg_info):
```
Comment (hayk-skydio) [inline setup.py:115]: docstring

File: setup.py [hunk 5]
```diff
@@ -211,10 +260,48 @@ def symforce_data_files() -> T.List[str]:
...
+    # Replace relative links with absolute, so images appear correctly on PyPI
+    readme = readme.replace(
+        "docs/static/images/",
+        f"https://raw.githubusercontent.com/symforce-org/symforce/{symforce_rev()}/docs/static/images/",
+    )
+
+    # Remove the DARK_MODE_ONLY tags
+    # See https://stackoverflow.com/a/1732454/2791611
+    readme = re.sub(
+        r"<!--\s*DARK_MODE_ONLY\s*-->((?!DARK_MODE_ONLY).)*<!--\s*/DARK_MODE_ONLY\s*-->",
```
Comment (hayk-skydio) [inline setup.py:296]: ah found it

File: setup.py [hunk 6]
```diff
@@ -224,7 +311,7 @@ def symforce_data_files() -> T.List[str]:
     keywords="python computer-vision cpp robotics optimization structure-from-motion motion-planning code-generation slam autonomous-vehicles symbolic-computation",
     license="Apache 2.0",
     license_file="LICENSE",
-    long_description=Path("README.md").read_text(),
+    long_description=fixed_readme(),
```
Comment (hayk-skydio) [inline setup.py:314]: this is some dedication, well done

File: third_party/symenginepy/setup.py [hunk 7]
```diff
@@ -61,9 +61,13 @@ def get_build_dir(dist):
      'options to cmake <var>:<type>=<value>'),
 ]
 
-def _process_define(arg):
-    (defs, one), = getattr(arg, 'define', None) or [('', '1')]
-    assert one == '1'
+def _process_define(arg, install=False):
```
Comment (hayk-skydio) [inline third_party/symenginepy/setup.py:64]: docstring

**Review by hayk-skydio — APPROVED** [review]

File: CMakeLists.txt (hunk 1 above)
Comment (aaron-skydio) [reply CMakeLists.txt:98]: Hmm possibly yeah - I don't think we'd want to be defining toolchains for users to use, it'd just be for our CI (if a user wants a toolchain they should define it, I don't _think_ it's correct for a library to define a toolchain?).  Not sure how that'd look for this wheel building process (i.e. how we'd pipe it through)

File: setup.py (hunk 3 above)
Comment (aaron-skydio) [reply setup.py]: Done

File: setup.py (hunk 4 above)
Comment (aaron-skydio) [reply setup.py:117]: Done

File: third_party/symenginepy/setup.py (hunk 7 above)
Comment (aaron-skydio) [reply third_party/symenginepy/setup.py:64]: Done

**Pull request closed without merging**